clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -202,6 +203,10 @@ public synchronized boolean updateRequested() {
         return this.needFullUpdate || this.needPartialUpdate;
     }
 
+    public synchronized void addClusterUpdateListener(ClusterResourceListener listener) {
+        this.clusterResourceListeners.maybeAdd(listener);
```

Comment:
Because of the way the maybeAddAll is used [here](https://github.com/philipnee/kafka/blob/aca2552244727668880fed700c0659f67e116a26/clients/src/main/java/org/apache/kafka/clients/ClientUtils.java#L260) (called with lists of objects that may not all implement the ClusterResourceListener, ex. ProducerInterceptor, ConsumerInterceptors, MetricsReporter, Deserializer.